scipy/spatial/distance.py

```diff
@@ -1421,10 +1481,6 @@ def dfun(u, v):
         elif mstr in ['minkowski', 'mi', 'm']:
             X = _convert_to_double(X)
             _distance_wrap.pdist_minkowski_wrap(X, dm, p)
-        elif mstr in ['wminkowski', 'wmi', 'wm', 'wpnorm']:
-            X = _convert_to_double(X)
-            w = _convert_to_double(np.asarray(w))
-            _distance_wrap.pdist_weighted_minkowski_wrap(X, dm, p, w)
```

Comment:
Removing this would break all code that is currently use the weighted version of minkowski. IMHO, it should possibly raise a deprecated warning and call the right function according to whether or not the weights are passed as args.